core-processor/src/executor.rs

```diff
@@ -51,94 +50,57 @@ use gear_core_backend::{
         ActorTerminationReason, BackendAllocSyscallError, BackendSyscallError, RunFallibleError,
         TerminationReason,
     },
-    memory::MemoryWrap,
     BackendExternalities,
 };
-use gear_lazy_pages_common::{GlobalsAccessConfig, LazyPagesWeights};
-use scale_info::{
-    scale::{self, Decode, Encode},
-    TypeInfo,
-};
-
-actor_system_error! {
-    /// Prepare memory error.
-    pub type PrepareMemoryError = ActorSystemError<ActorPrepareMemoryError, SystemPrepareMemoryError>;
-}
-
-// TODO: add this cases checks to Code::try_new in gear-core #3735
-/// Prepare memory error
-#[derive(Encode, Decode, TypeInfo, Debug, PartialEq, Eq, PartialOrd, Ord, derive_more::Display)]
-#[codec(crate = scale)]
-pub enum ActorPrepareMemoryError {
-    /// Stack end page, which value is specified in WASM code, cannot be bigger than static memory size.
-    #[display(fmt = "Stack end page {_0:?} is bigger then WASM static memory size {_1:?}")]
-    StackEndPageBiggerWasmMemSize(WasmPage, WasmPage),
-    /// Stack is not aligned to WASM page size
-    #[display(fmt = "Stack end addr {_0:#x} must be aligned to WASM page size")]
-    StackIsNotAligned(u32),
-}
 
-#[derive(Debug, Eq, PartialEq, derive_more::Display)]
-pub enum SystemPrepareMemoryError {
-    /// Mem size less than static pages num
-    #[display(fmt = "Mem size less than static pages num")]
-    InsufficientMemorySize,
-}
-
-/// Make checks that everything with memory goes well.
-fn check_memory(
-    static_pages: WasmPage,
+fn validate_memory_params(
```

Comment:
Couldn't find note comment, please add it anyway

Reply:
Yes, after some thinking I decide not to append it because this checks should be done by AllocationsContext::new, which should check consistency of provided params

Reply:
Then to be resolved in #3813, right?

Reply:
Ok append comment not to forget in future